src/main/java/org/kaazing/netx/http/internal/HttpURLConnectionHandler.java

```diff
@@ -252,11 +252,12 @@ public InputStream getInputStream() throws IOException {
             case HANDSHAKE_SENT:
                 input = socket.getInputStream();
 
-                BufferedReader reader = new BufferedReader(new InputStreamReader(input, "US-ASCII"));
-                String start = reader.readLine();
+//                BufferedReader reader = new BufferedReader(new InputStreamReader(input, "US-ASCII"));
```

Comment:
Suggest encapsulating the required buffer-only-to-end-of-line behavior into a `LineReader` class, such that this becomes:
```
LineReader reader = new LineReader(new InputStreamReader(input, "US-ASCII"));
```
and then the remaining changes are no longer needed.

Note: `LineReader.readLine(...)` should return `null` at `EOF` rather than the empty string for consistency with `BufferedReader`.